src/components/Calendar/index.js

```diff
@@ -29,38 +45,136 @@ class CalendarComponent extends Component {
+        this.handleKeyLeftPressed = this.handleKeyLeftPressed.bind(this);
+        this.handleKeyRightPressed = this.handleKeyRightPressed.bind(this);
+        this.handleKeyHomePressed = this.handleKeyHomePressed.bind(this);
+        this.handleKeyEndPressed = this.handleKeyEndPressed.bind(this);
+        this.handleKeyPageUpPressed = this.handleKeyPageUpPressed.bind(this);
+        this.handleKeyPageDownPressed = this.handleKeyPageDownPressed.bind(this);
+        this.handleKeyEnterPressed = this.handleKeyEnterPressed.bind(this);
+        this.handleKeyAltPageUpPressed = this.handleKeyAltPageUpPressed.bind(this);
+        this.handleKeyAltPageDownPressed = this.handleKeyAltPageDownPressed.bind(this);
+        this.keyHandlerMap = {
+            [UP_KEY]: this.handleKeyUpPressed,
+            [DOWN_KEY]: this.handleKeyDownPressed,
+            [LEFT_KEY]: this.handleKeyLeftPressed,
+            [RIGHT_KEY]: this.handleKeyRightPressed,
+            [HOME_KEY]: this.handleKeyHomePressed,
+            [END_KEY]: this.handleKeyEndPressed,
+            [PAGEUP_KEY]: this.handleKeyPageUpPressed,
+            [PAGEDN_KEY]: this.handleKeyPageDownPressed,
+            [SPACE_KEY]: this.handleKeyEnterPressed,
+            [ENTER_KEY]: this.handleKeyEnterPressed,
+        };
+        this.keyHandlerMapAlt = {
+            [PAGEUP_KEY]: this.handleKeyAltPageUpPressed,
+            [PAGEDN_KEY]: this.handleKeyAltPageDownPressed,
+        };
     }
 
-    componentDidUpdate(prevProps) {
+    componentDidUpdate(prevProps, prevState) {
         const { value: prevValue } = prevProps;
         const { value } = this.props;
         const normalizedDate = normalizeDate(value);
         if (formatDate(normalizeDate(prevValue)) !== formatDate(normalizedDate)) {
             this.updateCurrentMonth(normalizedDate);
+            this.updateFocusedDate(normalizedDate);
+        }
+
+        const { showFocusedDate: prevShowFocusedDate } = prevState;
+        const { showFocusedDate } = this.state;
+        if (showFocusedDate && !prevShowFocusedDate) {
+            this.tableRef.current.focus();
         }
     }
 
+    getContext() {
+        const { showFocusedDate, focusedDate } = this.state;
+        return {
+            showFocusedDate,
+            focusedDate,
+        };
+    }
+
+    moveFocusedDay(increment) {
+        const { minDate, maxDate } = this.props;
+        const { currentMonth, focusedDate } = this.state;
+        let nextFocusedDate = addDays(focusedDate, increment);
+        let nextFocusedMonth = currentMonth;
+
+        if (nextFocusedDate.getMonth() !== currentMonth.getMonth()) {
+            nextFocusedMonth = getFirstDayMonth(addMonths(currentMonth, Math.sign(increment)));
+        }
+
+        if (minDate && compareDates(nextFocusedDate, minDate) < 0) {
+            nextFocusedDate = minDate;
+            nextFocusedMonth = getFirstDayMonth(minDate);
+        } else if (maxDate && compareDates(nextFocusedDate, maxDate) > 0) {
```

Comment:
here the check for minDate and maxDate are preventing to stop when you get to max and min boundaries when they are undefined (not passed minDate and maxDate props)